source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems,
+       Nix community has chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is Open Source. 
+          It's only possible to cross-compile between ``aarch64-darwin`` and ``x86_64-darwin``.
```

Comment:
Technically only from `x86_64-darwin` to `aarch64-darwin`. Not sure if I'd mention this here though, as `aarch64-darwin` is far from mature at this point and cross compilation is not working at all in any channel  currently…

Reply:
> Technically only from x86_64-darwin to aarch64-darwin.

Is that because we're missing `pkgsCross.x86_64-darwin` or something else?

> Not sure if I'd mention this here though, as aarch64-darwin is far from mature at this point and cross compilation is not working at all in any channel currently…

Since the fix from @alyssais should work, the whole cross-compilation is not well supported but I'd like this tutorial to show that it's entirely feasible with enough maintainers. WDYT?

Reply:
> Is that because we're missing pkgsCross.x86_64-darwin or something else?

I'm not sure as to why exactly, but I can confirm that nixpkgs can't even cross compile from x86_64-darwin to x86_64-darwin due to an evaluation problem very early in `stdenv`.

> the whole cross-compilation is not well supported but I'd like this tutorial to show that it's entirely feasible with enough maintainers

Still, aarch64-linux for example has had more time to mature, accumulate fixes etc. Maybe not mentioning aarch64-darwin saves someone a bit of frustration?

Reply:
How about a disclaimer that we've just added support for it?

Reply:
I've added a disclaimer and move this further down.